File: sdks/dotnet/src/Resources/StockTransfersResource.cs
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class StockTransfersResource
    {
        private readonly PuxbayClient _client;
        public StockTransfersResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<StockTransfer>> ListAsync(int page = 1, string status = null)
        {
            string query = $"stock-transfers/?page={page}";
            if (!string.IsNullOrEmpty(status)) query += $"&status={status}";
            return await _client.GetAsync<PaginatedResponse<StockTransfer>>(query);
        }

        public async Task<StockTransfer> GetAsync(string transferId)
        {
            return await _client.GetAsync<StockTransfer>($"stock-transfers/{transferId}/");
        }

        public async Task<StockTransfer> CreateAsync(StockTransfer transfer)
        {
            return await _client.PostAsync<StockTransfer>("stock-transfers/", transfer);
        }

        public async Task<StockTransfer> CompleteAsync(string transferId)
        {
            return await _client.PostAsync<StockTransfer>($"stock-transfers/{transferId}/complete/", null);
        }
    }
}

File: sdks/php/src/Resources/StockTransfersResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\StockTransfer;
use Puxbay\Models\PaginatedResponse;

class StockTransfersResource extends BaseResource
{
    public function list(int $page = 1, ?string $status = null): PaginatedResponse
    {
        $endpoint = "stock-transfers/?page={$page}";
        if ($status) {
            $endpoint .= "&status={$status}";
        }
        $data = $this->get($endpoint);
        return PaginatedResponse::fromArray($data, fn($item) => StockTransfer::fromArray($item));
    }

    public function getTransfer(string $transferId): StockTransfer
    {
        $data = $this->get("stock-transfers/{$transferId}/");
        return StockTransfer::fromArray($data);
    }

    public function create(StockTransfer $transfer): StockTransfer
    {
        $data = $this->post('stock-transfers/', $transfer->toArray());
        return StockTransfer::fromArray($data);
    }

    public function complete(string $transferId): StockTransfer
    {
        $data = $this->post("stock-transfers/{$transferId}/complete/");
        return StockTransfer::fromArray($data);
    }
}
